### PR TITLE
Add misk-guice module containing former :misk:misk misk.inject package

### PR DESCRIPTION
**Pull request by jjestrel**
part of the work to pull out `misk-client`

### CODE REVIEW

File: misk-aws/src/main/kotlin/misk/cloud/aws/AwsEnvironmentModule.kt [hunk 1]
```diff
@@ -3,7 +3,7 @@ package misk.cloud.aws
 import com.google.inject.Provides
 import misk.environment.EnvVarLoader
 import misk.environment.ForEnvVars
-import misk.inject.KAbstractModule
+import misk.guice.KAbstractModule
```
Comment (swankjesse) [inline misk-aws/src/main/kotlin/misk/cloud/aws/AwsEnvironmentModule.kt]: I like the new module, but renaming the package seems like needless churn. Is this so we can have both concurrently?

File: misk-guice/build.gradle [hunk 2]
```diff
@@ -0,0 +1,14 @@
+buildscript {
+  dependencies {
+    classpath dep.kotlinNoArgPlugin
+  }
+}
+
+apply plugin: 'kotlin-jpa'
```
Comment (swankjesse) [inline misk-guice/build.gradle]: Necessary?

File: misk-aws/src/main/kotlin/misk/cloud/aws/AwsEnvironmentModule.kt (hunk 1 above)
Comment (jjestrel) [reply misk-aws/src/main/kotlin/misk/cloud/aws/AwsEnvironmentModule.kt]: changed back

File: misk-guice/build.gradle (hunk 2 above)
Comment (jjestrel) [reply misk-guice/build.gradle]: removed

**Comment by jjestrel** [comment]
The format changes are auto intellij changes

**Pull request merged**